### ISSUE TITLE
v1.3: Jobs models from disabled plugins result in inconsistent UI behavior

### ISSUE THREAD

**Issue by jathanism** [nautobot/nautobot#1563]
### Environment
* Python version:
* Nautobot version:

### Steps to Reproduce
0. Make sure a worker is running
1. Enable `example_plugin` (`PLUGINS = ["example_plugin"]` in `nautobot_config.py`)
2. Run migrations
3. Restart Nautobot to get Job models populated for each job provided by `example_plugin`
4. Navigate to Jobs list, then "Example logging" job
5. Observe that this job is "Installed = True", "Enabled = False"; edit the job, and set "Enabled = True"
6. Run the job. It should succeed.
7. Disable `example_plugin` in `nautobot_config.py`
8. Restart Nautobot again
9. Observe that job is still set to "Installed = True"
10. Click "Run/Schedule"

### Expected Behavior

Job should not run, displaying an error in UI.

### Observed Behavior

500 error with 

```
Environment:


Request Method: GET
Request URL: http://localhost:6789/extras/jobs/plugins-example_plugin-jobs-exampleloggingjob/run/

Django Version: 3.2.12
Python Version: 3.9.2
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'cacheops',
 'corsheaders',
 'django_filters',
 'django_jinja',
 'django_tables2',
 'django_prometheus',
 'mptt',
 'rest_framework',
 'social_django',
 'taggit',
 'timezone_field',
 'nautobot.core.apps.NautobotConstanceConfig',
 'nautobot.core',
 'django.contrib.admin',
 'django_celery_beat',
 'db_file_storage',
 'nautobot.circuits',
 'nautobot.dcim',
 'nautobot.ipam',
 'nautobot.extras',
 'nautobot.tenancy',
 'nautobot.users',
 'nautobot.utilities',
 'nautobot.virtualization',
 'django_rq',
 'drf_spectacular',
 'drf_spectacular_sidecar',
 'graphene_django',
 'health_check',
 'health_check.cache',
 'health_check.storage',
 'django_extensions',
 'nautobot.core.apps.ConstanceDatabaseAppConfig',
 'django_ajax_tables',
 'sentry_sdk',
 'crispy_forms']
Installed Middleware:
['django_prometheus.middleware.PrometheusBeforeMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'nautobot.core.middleware.ExceptionHandlingMiddleware',
 'nautobot.core.middleware.RemoteUserMiddleware',
 'nautobot.core.middleware.ExternalAuthMiddleware',
 'nautobot.core.middleware.ObjectChangeMiddleware',
 'django_prometheus.middleware.PrometheusAfterMiddleware']



Traceback (most recent call last):
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.9/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.9/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/django/views.py", line 67, in sentry_wrapped_callback
    return callback(request, *args, **kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.9/lib/python3.9/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/jathan/sandbox/src/nautobot/nautobot/utilities/views.py", line 94, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.9/lib/python3.9/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/Users/jathan/sandbox/src/nautobot/nautobot/extras/views.py", line 953, in get
    job_form = job_model.job_class().as_form(initial=normalize_querydict(request.GET))
  File "/Users/jathan/sandbox/src/nautobot/nautobot/extras/jobs.py", line 301, in as_form
    job_model = JobModel.objects.get_for_class_path(self.class_path)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.9/lib/python3.9/site-packages/django/utils/functional.py", line 61, in __get__
    return self.fget(cls)
  File "/Users/jathan/sandbox/src/nautobot/nautobot/extras/jobs.py", line 154, in class_path
    raise RuntimeError(

Exception Type: RuntimeError at /extras/jobs/plugins-example_plugin-jobs-exampleloggingjob/run/
Exception Value: Unknown/unexpected job file_path /Users/jathan/sandbox/src/nautobot/examples/example_plugin/example_plugin/jobs.py, should be one of /Users/jathan/.nautobot/jobs, /Users/jathan/.nautobot/git
```

**Comment by glennmatthews** [comment]
On the one hand, this is incorrect usage (after installing *or uninstalling* a plugin you should run `nautobot-server post_upgrade`), but on the other hand we should at least have code to catch and gracefully handle this issue. Thanks for finding and documenting this. :-)

**Comment by jathanism** [comment]
Just as a quick initial pass, catching the exception and printing an error message like this:

```python
diff --git a/nautobot/extras/views.py b/nautobot/extras/views.py
index 5c570b600..5e6e973ae 100644
--- a/nautobot/extras/views.py
+++ b/nautobot/extras/views.py
@@ -950,7 +950,12 @@ class JobView(ObjectPermissionRequiredMixin, View):
     def get(self, request, class_path=None, slug=None):
         job_model = self._get_job_model_or_404(class_path, slug)

-        job_form = job_model.job_class().as_form(initial=normalize_querydict(request.GET))
+        try:
+            job_form = job_model.job_class().as_form(initial=normalize_querydict(request.GET))
+        except RuntimeError:
+            job_form = None
+            messages.error(request, "Unable to run or schedule job: Job is not presently installed.")
+
         schedule_form = forms.JobScheduleForm(initial=normalize_querydict(request.GET))

         return render(
(END)
```

Results in this: 
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/138052/160939696-4219bc96-8338-4472-a82d-340a33e8ea04.png">

**Comment by jathanism** [comment]
Better:
<img width="1477" alt="image" src="https://user-images.githubusercontent.com/138052/160940521-f36c98bf-9216-4f0c-91ca-8b2b44bf960e.png">

**Comment by bryanculver** [comment]
Closed with #1590
